cmd/process-agent/collector.go

```diff
@@ -228,63 +282,43 @@ func (l *Collector) run(exit chan bool) error {
 	return nil
 }
 
-func (l *Collector) consumePayloads(payloads chan checkPayload, fwd forwarder.Forwarder, exit chan bool) {
+func (l *Collector) consumePayloads(results *api.WeightedQueue, fwd forwarder.Forwarder, exit chan struct{}) {
 	for {
-		select {
-		case payload := <-payloads:
-			if len(payloads) >= l.cfg.QueueSize {
-				log.Info("Expiring payload from in-memory queue.")
-				// Limit number of items kept in memory while we wait.
-				<-payloads
+		item, ok := results.Poll(exit)
```

Comment:
Can we add a comment specifying that this is a blocking call until new checks arrive for us to process?